src/wallet/wallet.js

```diff
@@ -53,12 +59,22 @@ export class Wallet {
       throw new Error('Hex encoded session key expected.')
     }
 
+    keypairs = Array.from([
+      signingKeypair.secret(),
+      ...keypairs
+    ]
+      .reduce((result, item) => {
+        result.add(item)
+        return result
+      }, new Set()))
+      .map(item => Keypair.fromSecret(item))
```

Comment:
looks like `.reduce()` part can be simplified to just `new Set([signingKeypair.secret(), ...keypairs])`